### ISSUE TITLE
Option to always derive `enum`s as if they were `sealed trait`s

### ISSUE THREAD

**Issue by mrdziuban** [sirthias/borer#710]
Hi, thank you for this great library!

I'm curious if you would consider adding an option to always derive `enum`s as if they were `sealed trait`s/`sealed abstract class`es? That is, even a simple `enum` with only singleton `case`s would still produce a JSON object and could be configured with `AdtEncodingStrategy`.

For example:

```scala
import io.bullet.borer.{Encoder, Json}
import io.bullet.borer.derivation.MapBasedCodecs.*

enum Foo derives Encoder {
  case Bar
}
Json.encode(Foo.Bar).toUtf8String // "Bar"

sealed trait Baz derives Encoder.All
object Baz {
  case object Quux extends Baz
}
Json.encode(Baz.Quux: Baz).toUtf8String // {"Quux":{}}
```

I would like `Json.encode(Foo.Bar)` to produce `{"Baz":{}}` so I can use `enum`s and `sealed trait`s interchangeably

**Comment by sirthias** [comment]
Thanks, @mrdziuban, for this proposal.
I was thinking about the distinction between `enum` instances and singleton objects myself.
Conceptionally they are pretty much the same but technically they are very different, which `case objects` defining their own type extending `Product0` while `enum` cases simply being instances of their `enum` type.

If we were to implement the option you are asking for it'd have to be opt-in configurable, since we can't break existing encodings. One way could be to add a `enumCasesAsProduct` flag to `AdtEncodingStrategy`.
Would sth like that work for you?

**Comment by mrdziuban** [comment]
Yeah a new config option on `AdtEncodingStrategy` would be great, thank you!
